### PR TITLE
PR1 API interfaces definition

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
Split M2 Shipping and Tax endpoints

NOTE: Shipping and Tax split feature is divided into 4 PRs after it was finished. The split is done by dev workflow, position in the code base, execution logic and interference with the previous code (put in the last PR). However, the PRs are not independent of each other, that is visible in failed unit tests. The unit tests will pass when all 4 PRs get merged, as they do in the integral version at https://github.com/BoltApp/bolt-magento2/pull/641. That's why an intermediate branch is created, feature/shipping-and-tax-spli, instead of merging them directly into master.

Fixes: https://boltpay.atlassian.net/browse/M2P-51
https://github.com/BoltApp/bolt-magento2/pull/641

#changelog API interfaces definition

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by danac-gs** [comment]
Renew

File: Api/Data/TaxDataInterface.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /**
+     * Get order tax result.
+     *
+     * @return \Bolt\Boltpay\Api\Data\TaxResultInterface
+     * @api
+     */
+    public function getTaxResult();
+
+    /**
+     * Set available shipping options.
```
Comment (matt-thomason) [inline Api/Data/TaxDataInterface.php:37]: Should be `Set available tax result`

File: Api/Data/TaxDataInterface.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Api\Data;
+
+/**
+ * Tax data interface. Tax response object fields.
+ *
+ * Get shipping options available.
```
Comment (matt-thomason) [inline Api/Data/TaxDataInterface.php:23]: Is this incorrect copy+paste?

File: Api/Data/TaxResultInterface.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Tax result interface. Subtotal amount field of the Tax response object.
+ *
+ * @api
+ */
+interface TaxResultInterface
+{
+    /**
+     * Get subtotaltotal tax amount.
```
Comment (matt-thomason) [inline Api/Data/TaxResultInterface.php:28]: Just subtotal, also for next function.

File: Api/ShippingInterface.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{
+    /**
+     * Get all available shipping methods.
+     *
+     * @api
+     * @param mixed $cart cart details
+     * @param mixed $shipping_address shipping address
+     * @return \Bolt\Boltpay\Api\Data\ShippingDataInterface
+     */
+    public function execute($cart, $shipping_address);
```
Comment (matt-thomason) [inline Api/ShippingInterface.php:36]: `$shippingAddress`?

File: Api/TaxInterface.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    /**
+     * Get tax for a given shipping option.
+     *
+     * @api
+     * @param mixed $cart cart details
+     * @param mixed $shipping_address shipping address
+     * @param mixed $shipping_option selected shipping option
+     * @return \Bolt\Boltpay\Api\Data\TaxDataInterface
+     */
+    public function execute($cart, $shipping_address, $shipping_option = null);
```
Comment (matt-thomason) [inline Api/TaxInterface.php:37]: `$shippingAddress, $shippingOption`?

File: Api/Data/TaxDataInterface.php (hunk 2 above)
Comment (danac-gs) [reply Api/Data/TaxDataInterface.php:23]: Yep. Fixing.

File: Api/ShippingInterface.php (hunk 4 above)
Comment (danac-gs) [reply Api/ShippingInterface.php:36]: The variable name is auto-generated by Magento REST API from Bolt request. It is snake_case in request so it has to stay.

File: Api/TaxInterface.php (hunk 5 above)
Comment (danac-gs) [reply Api/TaxInterface.php:37]: The variable names are auto-generated by Magento REST API from Bolt request. They are snake_case in the request so they have to stay that way.

File: Api/TaxInterface.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Api;
+
+/**
+ * Tax interface Tax endpoint.
+ *
+ * Get shipping methods using shipping address and cart details.
```
[the remark this answers is not in the document]
Comment (danac-gs) [reply Api/TaxInterface.php]: fixed

**Pull request merged**